Sources/WebDriver/Element.swift

```diff
@@ -85,6 +85,43 @@ public struct Element {
         if let notInteractableError = result.value { throw notInteractableError }
     }
 
+    /// Double clicks an element by id.
+    public func doubleClick(element: String, retryTimeout: TimeInterval? = nil) throws {
+        let request = Requests.SessionTouchDoubleClick(session: session.id, element: id)
+        let result = try poll(timeout: retryTimeout ?? session.defaultRetryTimeout) {
+            do {
+                // Immediately bubble most failures, only retry on element not interactable.
+                try webDriver.send(request)
```

Comment:
It would be great to introduce an extension method `sendWithRetry` that did this whole boilerplate:

`func sendWithRetry<R: Request>(_ request: R, timeout: TimeInterval, retryIf: (ErrorResponse) -> Bool)`

(but I can do it in a follow-up PR)

Reply:
I can run this in a seperate pr. I think it might fall as out of scope for when someone was looking back at prs and wanted to know how it was implemented.

Reply:
Yup no worries.